server/lib/routes/otp.js

```diff
@@ -39,26 +39,27 @@ async function otpRoutes(server) {
         )
         .get()
 
-      if (subscriptions.empty) {
+      if (subscription.empty) {
         log.error(`No subscription found for user ${userId}`)
-        return reply.code(404).send(`No subscription found for user ${userId}`)
+        return reply.notFound(`No subscription found for user ${userId}`)
       }
 
       const uniqueId = uniqid()
 
       return new Promise(() => {
-        const completeRequest = (error, otp) => {
+        const completeRequest = (error, otp, requestObj) => {
+          requestObj.delete()
```

Comment:
isn't this an async operation that we may want to wait?

Reply:
It is an async operation but I don't think we have to necessarily wait